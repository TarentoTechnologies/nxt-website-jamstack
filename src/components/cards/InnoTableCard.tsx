import React from "react";

import {
  bgRhodaminePurple,
  bgSpectrumBlue,
  bgVividGreen,
} from "../../styles/style-guide/ColorGuide.module.css";
import { innoTitle } from "../../styles/style-guide/Typography.module.css";
import { cardStyles, titleStyles } from "./InnoTableCard.module.css";

interface InnoTableCardProps {
  title: string;
  desc: string;
  bgColor: any;
  clickHandler?: (event: any) => void;
}

export const InnoTableCard = ({
  title,
  desc,
  bgColor,
  clickHandler,
}: InnoTableCardProps) => {
  const getBgColor = () => {
    switch (bgColor) {
      case "blue":
        return bgSpectrumBlue;
      case "green":
        return bgVividGreen;
      case "pink":
        return bgRhodaminePurple;
      default:
        return "";
    }
  };
  const bgColour = getBgColor();
  return (
    <div
      className={`col-md-4 ${bgColour} ${cardStyles}`}
      onClick={clickHandler}
      data-aos="fade-up"
      data-testid="inno-table-card"
    >
      <h2 className={`${titleStyles} ${innoTitle}`} data-aos="fade-up">
        {title}
      </h2>
      <p data-aos="fade-up">{desc}</p>
    </div>
  );
};
