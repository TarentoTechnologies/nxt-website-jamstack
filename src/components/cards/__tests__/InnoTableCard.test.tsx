import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { InnoTableCard } from "../InnoTableCard";

describe("Test if the InnoTableCard component is working as expected", () => {
  it("<InnoTableCard /> matches snapshot", () => {
    const component = render(
      <InnoTableCard
        title={"TECHNOLOGY"}
        desc={
          "Our tech stack is crafted compact to abide by our technology serving principles to build for the future!"
        }
        bgColor={"blue"}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<InnoTableCard /> renders without exploding", () => {
    const component = render(
      <InnoTableCard
        title={"TECHNOLOGY"}
        desc={
          "Our tech stack is crafted compact to abide by our technology serving principles to build for the future!"
        }
        bgColor={"blue"}
      />
    );
    expect(component.getByTestId("inno-table-card")).toBeInTheDocument();
  });
});
