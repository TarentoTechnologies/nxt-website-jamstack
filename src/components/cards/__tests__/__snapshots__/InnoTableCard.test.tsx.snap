// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the InnoTableCard component is working as expected <InnoTableCard /> matches snapshot 1`] = `
<div>
  <div
    class="col-md-4 bgSpectrumBlue cardStyles"
    data-aos="fade-up"
    data-testid="inno-table-card"
  >
    <h2
      class="titleStyles innoTitle"
      data-aos="fade-up"
    >
      TECHNOLOGY
    </h2>
    <p
      data-aos="fade-up"
    >
      Our tech stack is crafted compact to abide by our technology serving principles to build for the future!
    </p>
  </div>
</div>
`;
